spring-jms/src/test/java/org/springframework/jms/core/JmsClientTests.java

```diff
@@ -121,24 +121,28 @@ void convertAndSendPayloadName() throws JMSException {
 	}
 
 	@Test
-	void convertAndSendPayloadAndHeaders() {
+	void convertAndSendPayloadAndHeaders() throws JMSException {
 		Destination destination = new Destination() {};
 		Map<String, Object> headers = new HashMap<>();
 		headers.put("foo", "bar");
 
 		this.jmsClient.destination(destination).send("Hello", headers);
 		verify(this.jmsTemplate).send(eq(destination), this.messageCreator.capture());
 		assertTextMessage(this.messageCreator.getValue()); // see createTextMessage
+		TextMessage jmsMessage = createTextMessage(this.messageCreator.getValue());
+		assertThat(jmsMessage.getStringProperty("foo")).isEqualTo("bar");
 	}
 
 	@Test
-	void convertAndSendPayloadAndHeadersName() {
+	void convertAndSendPayloadAndHeadersName() throws JMSException {
 		Map<String, Object> headers = new HashMap<>();
 		headers.put("foo", "bar");
 
 		this.jmsClient.destination("myQueue").send("Hello", headers);
 		verify(this.jmsTemplate).send(eq("myQueue"), this.messageCreator.capture());
 		assertTextMessage(this.messageCreator.getValue()); // see createTextMessage
+		TextMessage jmsMessage = createTextMessage(this.messageCreator.getValue());
+		assertThat(jmsMessage.getStringProperty("foo")).isEqualTo("bar");
```

Comment:
That's what `assertTextMessage()` already does.

Please revert this change, including the `throws` declaration.